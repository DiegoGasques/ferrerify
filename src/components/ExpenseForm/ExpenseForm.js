import React, { useState } from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

const ExpenseForm = ({ description, amount, note, createdAt, onSubmit }) => {
  const [_description, setDescription] = useState(description || "");
  const [_note, setNote] = useState(note || "");
  const [_amount, setAmount] = useState(
    amount ? (amount / 100).toString() : ""
  );
  const [_createdAt, setCreatedAt] = useState(
    createdAt ? moment(createdAt) : moment()
  );
  const [_focused, setFocused] = useState(false);
  const [_error, setError] = useState("");

  const handleChange = ({ target: { name, value } }) => {
    if (name === "description") return setDescription(value);
    if (name === "note") return setNote(value);
    if (name === "amount") {
      if (!value || value.match(/^\d{1,}(\.\d{0,2})?$/)) {
        return setAmount(value);
      }
    }
  };

  const handleDateChange = createdAt => {
    if (createdAt) {
      setCreatedAt(createdAt);
    }
  };

  const handleSubmit = e => {
    e.preventDefault();

    if (!_description || !_amount) {
      setError("Please provide description and amount.");
    } else {
      setError("");
      onSubmit({
        description: _description,
        note: _note,
        amount: parseFloat(_amount, 10) * 100,
        createdAt: _createdAt.valueOf()
      });
    }
  };

  return (
    <div>
      {_error && <p>{_error}</p>}
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          name="description"
          placeholder="Description"
          autoFocus
          value={_description}
          onChange={handleChange}
        />
        <input
          type="text"
          name="amount"
          placeholder="Ex: 10, 10.00"
          value={_amount}
          onChange={handleChange}
        />
        <SingleDatePicker
          date={_createdAt}
          onDateChange={handleDateChange}
          focused={_focused}
          onFocusChange={setFocused}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
        <textarea
          name="note"
          placeholder="Add a note for your expense (optional)"
          value={_note}
          onChange={handleChange}
        />
        <button>Add Expense</button>
      </form>
    </div>
  );
};

export default ExpenseForm;
