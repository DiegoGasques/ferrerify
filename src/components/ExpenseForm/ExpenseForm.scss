.ExpenseForm {
  display: flex;
  flex-direction: column;
  padding: $m-size;

  .error {
    background: lighten(red, 15%);
    color: white;
    padding: $s-size $m-size;
    margin-bottom: $m-size;
  }

  form {
    display: flex;
    flex-direction: column;

    .row {
      display: flex;
      margin-bottom: $s-size;
      width: 100%;
      @media (max-width: 700px) {
        flex-direction: column;
        .formGroup {
          margin-bottom: $m-size;
        }
      }
    }

    .formGroup {
      flex-grow: 1;
      display: flex;
      flex-direction: column;
    }

    .first {
      width: 60%;
      margin-right: $s-size;
      @media (max-width: 700px) {
        width: 100%;
      }
    }

    .third {
      @media (max-width: 700px) {
        width: 100%;
      }
    }

    input {
      flex-grow: 1;
      margin-right: $s-size;
      padding: $s-size;
      font-size: $h4;
      color: darken($light-grey, 34%);
      border: 1px solid $light-grey;
    }

    input:active {
      outline: none;
      border-bottom: 3px solid $primary;
    }

    input:focus {
      @extend input:active;
    }

    textarea {
      @extend input;
    }

    textarea {
      height: 10rem;
    }
  }

  button {
    width: 20rem;
    padding: $m-size;
    text-transform: uppercase;
    background: rgb(243, 9, 106);
    color: $snow;
    font-size: $h5;
    border: none;
    font-weight: bold;
    letter-spacing: 0.2rem;
  }

  button:hover {
    box-shadow: $btn-shadow-hover;
  }

  button:focus {
    outline: none;
  }

  button:active {
    outline: none;
    box-shadow: $btn-shadow-active;
  }

  label {
    color: $black;
  }
}
