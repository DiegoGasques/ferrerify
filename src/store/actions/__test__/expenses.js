import { addExpense, removeExpense, editExpense } from "../expenses.actions.js";

it("Test for addExpense action creators correct output", () => {
  const newExpense = {
    description: "New expense",
    note: "This is a big expense",
    createdAt: 1000000,
    amount: 15000
  };
  const action = addExpense(newExpense);

  expect(action).toEqual({
    type: "ADD_EXPENSE",
    expense: {
      ...newExpense,
      id: expect.any(String)
    }
  });
});

it("Test for removeExpense action creators correct output", () => {
  const id = "123";
  const action = removeExpense(id);
  expect(action).toEqual({
    type: "REMOVE_EXPENSE",
    id
  });
});

it("Test for editExpense action creators correct output", () => {
  const id = "123";
  const updates = { note: "The updated note" };
  const action = editExpense(id, updates);
  expect(action).toEqual({
    type: "EDIT_EXPENSE",
    id,
    updates
  });
});
