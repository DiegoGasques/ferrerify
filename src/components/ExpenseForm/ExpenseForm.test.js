import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ExpenseForm from "./ExpenseForm";
import { expensesArr } from "../../fixtures/expenses";

describe("[COMPONENT] ExpenseForm", () => {
  it("Tests for changes in the rendered component with default props", () => {
    const wrapper = shallow(<ExpenseForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Tests for changes in the rendered component with expense passed in", () => {
    const wrapper = shallow(<ExpenseForm {...expensesArr[0]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Tests for form submition event that should set error to state", () => {
    const wrapper = shallow(<ExpenseForm />);
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find("form").simulate("submit", {
      preventDefault: () => {}
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Tests for handleChange call changing the description", () => {
    const wrapper = shallow(<ExpenseForm />);
    expect(toJson(wrapper.find('input[name="description"]'))).toMatchSnapshot();

    const value = "new description";
    wrapper.find('input[name="description"]').simulate("change", {
      target: { value, name: "description" }
    });
    expect(toJson(wrapper.find('input[name="description"]'))).toMatchSnapshot();
  });

  it("Tests for handleChange call changing the note", () => {
    const wrapper = shallow(<ExpenseForm />);
    expect(toJson(wrapper.find('textarea[name="note"]'))).toMatchSnapshot();

    const value = "new note!!!!!!!";
    wrapper.find('textarea[name="note"]').simulate("change", {
      target: { value, name: "note" }
    });
    expect(toJson(wrapper.find('textarea[name="note"]'))).toMatchSnapshot();
  });

  it("Tests for handleChange call changing the amount", () => {
    const wrapper = shallow(<ExpenseForm />);
    expect(toJson(wrapper.find('input[name="amount"]'))).toMatchSnapshot();

    const value = "63278233.56";
    wrapper.find('input[name="amount"]').simulate("change", {
      target: { value, name: "amount" }
    });
    expect(toJson(wrapper.find('input[name="amount"]'))).toMatchSnapshot();
  });

  it("Tests for handleChange call when amount value passed is invalid", () => {
    const wrapper = shallow(<ExpenseForm />);
    expect(toJson(wrapper.find('input[name="amount"]'))).toMatchSnapshot();

    const value = "633,";
    wrapper.find('input[name="amount"]').simulate("change", {
      target: { value, name: "amount" }
    });
    expect(toJson(wrapper.find('input[name="amount"]'))).toMatchSnapshot();
  });
});
