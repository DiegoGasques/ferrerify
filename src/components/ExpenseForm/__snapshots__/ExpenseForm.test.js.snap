// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] ExpenseForm Tests for changes in the rendered component with default props 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      name="description"
      onChange={[Function]}
      placeholder="Description"
      type="text"
      value=""
    />
    <input
      name="amount"
      onChange={[Function]}
      placeholder="Ex: 10, 10.00"
      type="text"
      value=""
    />
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      horizontalMonthPadding={13}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonthElement={null}
      renderMonthText={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <textarea
      name="note"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      value=""
    />
    <button>
      Add Expense
    </button>
  </form>
</div>
`;

exports[`[COMPONENT] ExpenseForm Tests for changes in the rendered component with expense passed in 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      name="description"
      onChange={[Function]}
      placeholder="Description"
      type="text"
      value="Rent"
    />
    <input
      name="amount"
      onChange={[Function]}
      placeholder="Ex: 10, 10.00"
      type="text"
      value="1500"
    />
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      horizontalMonthPadding={13}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonthElement={null}
      renderMonthText={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <textarea
      name="note"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      value="Monthly payment"
    />
    <button>
      Add Expense
    </button>
  </form>
</div>
`;

exports[`[COMPONENT] ExpenseForm Tests for form submition event that should set error to state 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      name="description"
      onChange={[Function]}
      placeholder="Description"
      type="text"
      value=""
    />
    <input
      name="amount"
      onChange={[Function]}
      placeholder="Ex: 10, 10.00"
      type="text"
      value=""
    />
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      horizontalMonthPadding={13}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonthElement={null}
      renderMonthText={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <textarea
      name="note"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      value=""
    />
    <button>
      Add Expense
    </button>
  </form>
</div>
`;

exports[`[COMPONENT] ExpenseForm Tests for form submition event that should set error to state 2`] = `
<div>
  <p>
    Please provide description and amount.
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      name="description"
      onChange={[Function]}
      placeholder="Description"
      type="text"
      value=""
    />
    <input
      name="amount"
      onChange={[Function]}
      placeholder="Ex: 10, 10.00"
      type="text"
      value=""
    />
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      horizontalMonthPadding={13}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonthElement={null}
      renderMonthText={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <textarea
      name="note"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      value=""
    />
    <button>
      Add Expense
    </button>
  </form>
</div>
`;

exports[`[COMPONENT] ExpenseForm Tests for handleChange call changing the amount 1`] = `
<input
  name="amount"
  onChange={[Function]}
  placeholder="Ex: 10, 10.00"
  type="text"
  value=""
/>
`;

exports[`[COMPONENT] ExpenseForm Tests for handleChange call changing the amount 2`] = `
<input
  name="amount"
  onChange={[Function]}
  placeholder="Ex: 10, 10.00"
  type="text"
  value="63278233.56"
/>
`;

exports[`[COMPONENT] ExpenseForm Tests for handleChange call changing the description 1`] = `
<input
  autoFocus={true}
  name="description"
  onChange={[Function]}
  placeholder="Description"
  type="text"
  value=""
/>
`;

exports[`[COMPONENT] ExpenseForm Tests for handleChange call changing the description 2`] = `
<input
  autoFocus={true}
  name="description"
  onChange={[Function]}
  placeholder="Description"
  type="text"
  value="new description"
/>
`;

exports[`[COMPONENT] ExpenseForm Tests for handleChange call changing the note 1`] = `
<textarea
  name="note"
  onChange={[Function]}
  placeholder="Add a note for your expense (optional)"
  value=""
/>
`;

exports[`[COMPONENT] ExpenseForm Tests for handleChange call changing the note 2`] = `
<textarea
  name="note"
  onChange={[Function]}
  placeholder="Add a note for your expense (optional)"
  value="new note!!!!!!!"
/>
`;

exports[`[COMPONENT] ExpenseForm Tests for handleChange call when amount value passed is invalid 1`] = `
<input
  name="amount"
  onChange={[Function]}
  placeholder="Ex: 10, 10.00"
  type="text"
  value=""
/>
`;

exports[`[COMPONENT] ExpenseForm Tests for handleChange call when amount value passed is invalid 2`] = `
<input
  name="amount"
  onChange={[Function]}
  placeholder="Ex: 10, 10.00"
  type="text"
  value=""
/>
`;
